.clang-tidy

```diff
@@ -26,9 +26,18 @@ Checks:     '
             performance-*,
             portability-*,
             readability-*,
+            -bugprone-exception-escape,
+            -bugprone-macro-parentheses,
+            -bugprone-too-small-loop-variable,
```

Comment:
It complains about uint uin8_t or uint16_t when the limit on the loop is a container's .size() (size_t). We usually do this when we know the maximum size of the container though, so propose leaving disabled unless someone wants to refactor all of those for loops. Seems unnecessary.